experimental/packages/otlp-grpc-exporter-base/src/grpc-exporter-transport.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// NOTE: do not change these type imports to actual imports. Doing so WILL break `@opentelemetry/instrumentation-http`,
+// as they'd be imported before the http/https modules can be wrapped.
+import type { Metadata, ServiceError, ChannelCredentials } from '@grpc/grpc-js';
+import { ExportResponse } from './export-response';
+import { IExporterTransport } from './exporter-transport';
+
+const GRPC_COMPRESSION_NONE = 0;
+const GRPC_COMPRESSION_GZIP = 2;
+
+function toGrpcCompression(compression: 'gzip' | 'none'): number {
+  if (compression === 'none') {
+    return GRPC_COMPRESSION_NONE;
+  } else if (compression === 'gzip') {
+    return GRPC_COMPRESSION_GZIP;
+  }
+  return GRPC_COMPRESSION_NONE;
+}
+
+export function createInsecureCredentials(): ChannelCredentials {
+  const {
+    credentials,
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+  } = require('@grpc/grpc-js');
```

Comment:
I think a comment explaining why these are `require()` not import might be a good idea for future spelunkers of this code

Reply:
you mean why they're required here instead of on the top-level? Makes sense, I'll add a comment.

Reply:
applied in [231259e](https://github.com/open-telemetry/opentelemetry-js/pull/4432/commits/231259e3e1d212b5ad426c90675a2f3a6a3acdf0) and [638bc6f](https://github.com/open-telemetry/opentelemetry-js/pull/4432/commits/638bc6f1984ded569b957686e2ffbfd946b83199) :slightly_smiling_face: